Makefile

```diff
@@ -23,12 +23,12 @@ lint-check:
 
 test:
 	@echo "--- 🧪 Running tests ---"
-	pytest -n auto -m "not (test_datasets or leaderboard_stability)"
+	uv run pytest -n auto -m "not (test_datasets or leaderboard_stability)"
```

Comment:
we need to add the extra and group flags to all `uv run` where needed (it will install the required dependencies)